### PR TITLE
Fix customized product being added to cart, instead of standard one

### LINKED ISSUE #16682: Product is added to cart with customization, when it should not
**To Reproduce**
1. Set up an optional customization field on a product.
2. Go to this product in FO.
3. Customize it, save it, add it to cart.
4. Hit continue shopping.
5. Product page will refresh, saved customization will disappear visually.
6. Add the product to cart again. It should be added without customization, but it is added WITH the previous customization.

**The cause**
The input field is not updated on prestashop updatedProduct.
`<input type="hidden" name="id_customization" value="9" id="product_customization_id">`

**Additional information**
PrestaShop version: 1.7.6.2

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | https://github.com/PrestaShop/PrestaShop/issues/16682
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes https://github.com/PrestaShop/PrestaShop/issues/16682
| How to test?  | Make a product with customization. Customize it. Add it to cart. Customization disappears from product page. Click Add to cart again, you will add second customized piece to cart, instead of normal product. Happens because id_customization is not updated in it's hidden input. My PR fixes this.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/22145)

ping @kpodemski and @NeOMakinG - Guys, please, edit the selector to your liking. This works, but I am not a JS guy. Thanks! :-)

### CODE REVIEW

File: themes/_core/js/product.js [hunk 1]
```diff
@@ -220,6 +220,7 @@ function updateProduct(event, eventType, updateUrl) {
         $(prestashop.selectors.product.customization)
           .first()
           .replaceWith(data.product_customization);
+        $('.product-actions input[name=id_customization]').val(0);
```
Comment (Progi1984) [inline themes/_core/js/product.js]:
```suggestion
        $(`$(prestashop.selectors.product.actions) input[name=id_customization]`).val(0);
```

Comment (Progi1984) [reply themes/_core/js/product.js]: @NeOMakinG Should we add a new selector ?

**Comment by Hlavtox** [comment]
@Progi1984 I think there is no need to add another selector. This is perfectly fine, thank you man. 👍

File: themes/_core/js/product.js (hunk 1 above)
Comment (NeOMakinG) [reply themes/_core/js/product.js]: Of course it should be added to the selector's map!

Comment (Progi1984) [reply themes/_core/js/product.js]: Thanks @NeOMakinG 

So @Hlavtox could you add a new selector for this input (prestashop.selectors.product.input_customization for example) ?

**Comment by Hlavtox** [comment]
@NeOMakinG @Progi1984 I added a new selector and used it in the code. Is it OK now? 🚀

File: themes/_core/js/selectors.js [hunk 2]
```diff
@@ -43,6 +43,8 @@ prestashop.selectors = {
       '.quickview .product-add-to-cart, .page-product:not(.modal-open) .row .product-add-to-cart, .page-product:not(.modal-open) .product-container .product-add-to-cart',
     prices:
       '.quickview .product-prices, .page-product:not(.modal-open) .row .product-prices, .page-product:not(.modal-open) .product-container .product-prices',
+    input_customization:
```
Comment (NeOMakinG) [inline themes/_core/js/selectors.js]:
```suggestion
    inputCustomization:
```

File: themes/_core/js/product.js [hunk 3]
```diff
@@ -220,6 +220,7 @@ function updateProduct(event, eventType, updateUrl) {
         $(prestashop.selectors.product.customization)
           .first()
           .replaceWith(data.product_customization);
+        $(prestashop.selectors.product.input_customization).val(0);
```
Comment (NeOMakinG) [inline themes/_core/js/product.js]:
```suggestion
        $(prestashop.selectors.product.inputCustomization).val(0);
```

**Review by NeOMakinG — APPROVED** [review]
Just commit to use camelCase, lgtm now

**Review by PierreRambaud — APPROVED** [review]

**Comment by SD1982** [comment]
LGTM ! Thanks @Hlavtox !!

**Before PR**
**cart**
image
**order confirmation**
image
**order detail**
image
**invoice**
image




**After PR**
**cart**
image
**order confirmation**
image
**order detail**
image
**invoice**
image

Also verified in FO order detail
Also verified in order confirmation mail
Tested also with file customistion ( pdf, png, jpeg)
For this point the initial behaviour was different it seems that when the product was added the second time ( step 6 of #16682) the customisation was completely deleted => your PR fix it too :+1: 
Tested with both customisation ( file + text ) simultanously
Tested with 2 identical product but each with a different customisation

**Comment by Progi1984** [comment]
Thanks @Hlavtox & @SD1982

**Pull request merged**